test/unit/lib/migration-chunks/validation/editor-layout.spec.js

```diff
@@ -181,4 +181,305 @@ describe('editor layout plan validation', function () {
+            message: 'Field "foo" cannot be moved: missing field group pivot',
+            details: {
+              step: {
+                type: 'contentType/moveFieldInEditorLayout',
+                meta: { contentTypeInstanceId: 'contentType/page/0' },
+                payload: {
+                  contentTypeId: 'page',
+                  fieldId: 'foo',
+                  fieldGroupId: undefined,
+                  movement: { direction: 'afterFieldGroup', pivot: undefined }
+                }
+              }
+            }
+          }
+        ]);
+      });
+    });
+    describe('when pivot field group does not exist', () => {
+      it('returns an error', async () => {
+        const errors = await validateChunks(function (migration) {
+          const Page = migration.editContentType('page');
+          const editorLayout = Page.editEditorLayout();
+
+          editorLayout
+            .moveField('foo').afterFieldGroup('not_existing_group');
+        }, testCts, [], testEis);
+
+        expect(errors).to.eql([
+          {
+            type: 'InvalidAction',
+            message: 'Field "foo" cannot be moved: pivot field group "not_existing_group" does not exist',
+            details: {
+              step: {
+                type: 'contentType/moveFieldInEditorLayout',
+                meta: { contentTypeInstanceId: 'contentType/page/0' },
+                payload: {
+                  contentTypeId: 'page',
+                  fieldId: 'foo',
+                  fieldGroupId: undefined,
+                  movement: { direction: 'afterFieldGroup', pivot: 'not_existing_group' }
+                }
+              }
+            }
+          }
+        ]);
+      });
+    });
+    describe('when destination field group does not exist', () => {
+      it('returns an error', async () => {
+        const errors = await validateChunks(function (migration) {
+          const Page = migration.editContentType('page');
+          const editorLayout = Page.editEditorLayout();
+
+          editorLayout
+            .moveField('foo').toTheTopOfFieldGroup('not_existing_group');
+        }, testCts, [], testEis);
+
+        expect(errors).to.eql([
+          {
+            type: 'InvalidAction',
+            message: 'Field "foo" cannot be moved: destination group "not_existing_group" does not exist',
+            details: {
+              step: {
+                type: 'contentType/moveFieldInEditorLayout',
+                meta: { contentTypeInstanceId: 'contentType/page/0' },
+                payload: {
+                  contentTypeId: 'page',
+                  fieldId: 'foo',
+                  fieldGroupId: undefined,
+                  movement: { direction: 'toTheTopOfFieldGroup', pivot: 'not_existing_group' }
+                }
+              }
+            }
+          }
+        ]);
+      });
+    });
+    describe('when destination field group is not provided', () => {
+      it('succeeds', async () => {
```

Comment:
Can we make a more suitable test case out of it and check that it's moving inside the group where it already is?

Reply:
We do that in integration tests, here we can only check intent validation. I can add that scenario to the integration case